### DISCUSSION TITLE
Relationships in 2.0 seem to use Query

### DISCUSSION THREAD

**Discussion by ShmuelTreiger** [sqlalchemy/sqlalchemy#6716] [Usage Questions]
So when I went about updating all my queries into select statements, I also figured I would need to add `.scalars()` to my relationships. When I do so, I uniformly get the error `AttributeError: 'Query' object has no attribute 'scalars'`. Is there some way to switch relationships into using Select instead of Query? I've looked through the docs, but I can't find anywhere that they talk about something like that.

It matters somewhat, because I use custom Select and Query objects to implement custom where clauses. If I can't migrate relationships to Select then I will have to keep/maintain my Query object in addition to the Select object.

(Sorry, I think this is my last question)

**Comment by CaselIT** [comment]
I don't follow what you are trying to do.
Can you post an example?

**Reply by zzzeek** [reply]
I have a guess what they are doing but I feel badly that I have guessed it, because the description here really needs to include proper detail.
